lib/spotify/types.ts

```diff
@@ -25,3 +25,25 @@ export function isUserInfoResponse(res: any): res is UserInfoResponse {
     'uri' in res
   );
 }
+
+type TopSongs = {
+  'name': string,
+  'href': string, 
+  'uri': string, 
+  'popularity': number, 
+  'artists': Array<string>,
+};
+
+export type TopSongsResponse = {
+  [key: string]: TopSongs,
+};
+
+export function isTopSongsResponse(res: any): res is TopSongsResponse {
+  return (
+    'name' in res &&
+    'href' in res &&
+    'uri' in res &&
+    'popularity' in res &&
+    'artists' in res
+  );
```

Comment:
does this work?? we are trying to see if the response is in the form 

```ts
{
  [id: string]: {
    name: string,
    ...
  }
}
```

you are currently looking for this: 

```
{
  name: string,
  href: string,
  ...
}
```